src/hooks/usePACState.ts

```diff
@@ -0,0 +1,60 @@
+import { useK8sWatchResource } from '@openshift/dynamic-plugin-sdk-utils';
+import { PipelineRunLabel, PipelineRunType } from '../consts/pipelinerun';
+import { PipelineRunGroupVersionKind } from '../models';
+import { ComponentKind } from '../types';
+import { PipelineRunKind } from '../types/pipeline-run';
+import { isPACEnabled, SAMPLE_ANNOTATION } from '../utils/component-utils';
+
+export enum PACState {
+  sample,
+  disabled,
+  requested,
+  pending,
+  ready,
+  loading,
+}
+
+const usePACState = (component: ComponentKind) => {
+  const pacEnabled = isPACEnabled(component);
+  const pacDone = isPACEnabled(component, true);
+
+  // TODO need a better way to identify if the PR has merged on the main branch
+  const [pipelineBuildRuns, pipelineBuildRunsLoaded] = useK8sWatchResource<PipelineRunKind[]>(
+    pacEnabled
+      ? {
+          groupVersionKind: PipelineRunGroupVersionKind,
+          isList: true,
+          namespace: component.metadata.namespace,
+          selector: {
+            matchLabels: {
+              [PipelineRunLabel.PIPELINE_TYPE]: PipelineRunType.BUILD,
+              [PipelineRunLabel.COMPONENT]: component.metadata.name,
+            },
+            matchExpressions: [
+              { key: PipelineRunLabel.PULL_REQUEST_NUMBER_LABEL, operator: 'DoesNotExist' },
+              {
+                key: PipelineRunLabel.COMMIT_USER_LABEL,
+                value: 'appstudio-staging-ci__bot',
+                operator: 'NotEquals',
+              },
```

Comment:
```suggestion
              {
                key: PipelineRunLabel.COMMIT_USER_LABEL,
                value: 'appstudio-staging-ci__bot',
                operator: 'NotEquals',
              },
               {
                key: PipelineRunLabel.COMMIT_USER_LABEL,
                operator: 'Exists',
              },
```

I think we should also add if PAC commit exists as well to ensure the plrs are pac based.

I created a component with simple flow, and then upgraded to advanced flow expecting to see `Pull reqeuest sent` label but I am seeing `Custom build` in this case. The matchExpressions conditions matches the non-pac pipelinerun as well. 


<img width="1481" alt="image" src="https://user-images.githubusercontent.com/9964343/216976838-e3b56eb0-2798-4afd-ae58-ad9be20a53a8.png">